osu.Game/Screens/Select/BeatmapDetails.cs

```diff
@@ -0,0 +1,430 @@
+                    Width = 0.6f,
+                    Direction = FillDirection.Vertical,
+                    Spacing = new Vector2(0, 15),
+                    Padding = new MarginPadding(10) { Top = 0 },
+                    Children = new Drawable[]
+                    {
+                        new Container
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Children = new Drawable[]
+                            {
+                                new Box
+                                {
+                                    RelativeSizeAxes = Axes.Both,
+                                    Colour = Color4.Black,
+                                    Alpha = 0.5f,
+                                },
+                                new FillFlowContainer
+                                {
+                                    RelativeSizeAxes = Axes.X,
+                                    AutoSizeAxes = Axes.Y,
+                                    Direction = FillDirection.Vertical,
+                                    Spacing = new Vector2(0,10),
+                                    Padding = new MarginPadding(15) { Top = 25 },
+                                    Children = new []
+                                    {
+                                        circleSize = new DifficultyRow("Circle Size", 7),
+                                        drainRate = new DifficultyRow("HP Drain"),
+                                        overallDifficulty = new DifficultyRow("Accuracy"),
+                                        approachRate = new DifficultyRow("Approach Rate"),
+                                        stars = new DifficultyRow("Star Diffculty"),
+                                    },
+                                },
+                            },
+                        },
+                        ratingsContainer = new Container
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Alpha = 0,
+                            AlwaysPresent = true,
+                            Children = new Drawable[]
+                            {
+                                new Box
+                                {
+                                    RelativeSizeAxes = Axes.Both,
+                                    Colour = Color4.Black,
+                                    Alpha = 0.5f,
+                                },
+                                new FillFlowContainer
+                                {
+                                    RelativeSizeAxes = Axes.X,
+                                    AutoSizeAxes = Axes.Y,
+                                    Direction = FillDirection.Vertical,
+                                    Padding = new MarginPadding(15) { Top = 25, Bottom = 0 },

```

Comment:
this is really confusing. just expand it out rather than specifying at ctor *and* obj initialiser.